### PR TITLE
chore(deps)_: use gradle plugin to get deps

### PR DESCRIPTION
**Pull request by siddarthkay**
related issue : https://github.com/status-im/status-mobile/issues/15447

## Summary

This PR makes use of https://github.com/gradle/github-dependency-graph-gradle-plugin to generate deps so that we may get rid of the AWK script that parses `gradle` output to figure out `gradle` dependencies.

credits to @mendelskiv93 for doing initial research on this dependency generator plugin.

We still miss a few dependencies and are not completely able to get rid of the hack list step just yet.

I also moved `react-native-share` out of  `pluginManagement` block in `android/settings.gradle` because it does not belong there.

## Testing notes
not needed since we only change fetching build dependencies in this PR

## Platforms
- Android

status: ready

### CODE REVIEW

File: nix/deps/gradle/get_deps.sh [hunk 1]
```diff
@@ -15,23 +16,14 @@ GRADLE_LOG_FILE='/tmp/gradle.log'
...
+# remove when new architecture is enabled
+jq -r '.[].dependency | select(startswith("org.webkit:android-jsc") | not)' \
+"${GRADLE_REPORTS_DIR}/dependency-resolution.json" > "${GRADLE_REPORTS_DIR}/filtered-deps.txt"
 
-./gradlew --no-daemon --console plain \
-    "${BUILD_DEPS[@]}" \
-    "${NORMAL_DEPS[@]}" \
-    | tee "${GRADLE_LOG_FILE}" \
-    | awk -f "${AWK_SCRIPT}"
+cat "${GRADLE_REPORTS_DIR}/filtered-deps.txt"
```
Comment (jakubgs) [inline nix/deps/gradle/get_deps.sh]: Why put it into a file if we just print it afterwards?

**Review by jakubgs — COMMENTED** [review]
Promising.

File: nix/deps/gradle/get_deps.sh (hunk 1 above)
Comment (siddarthkay) [reply nix/deps/gradle/get_deps.sh]: Thanks! I was overcomplicating it, just `jq` was enough to do the filtering and print it.

**Review by jakubgs — APPROVED** [review]
My beautiful `nix/deps/gradle/gradle_parser.awk` script... you will be missed.

**Comment by seanstrom** [comment]
Nice stuff 🙌 

I've tried building the android app with a simulator using `make run-android`, and it all seems to build fine on my side. Is there anything else I should try to run for these changes?

**Review by ilmotta — APPROVED** [review]

**Comment by siddarthkay** [comment]
Thanks for testing locally.
> Is there anything else I should try to run for these changes?

No that should be enough!

**Pull request merged**